### ISSUE TITLE
I ported your example over to wai-routes

### ISSUE THREAD

**Issue by ajnsit** [parambirs/hello-scotty#1]
Hi,

I rewrote your example in wai-routes, just to gain perspective on how easy/difficult it is for beginners to get started with wai-routes.

https://github.com/ajnsit/wai-routes/blob/master/examples/kitchen/src/Main.hs

I welcome feedback on how to make wai-routes easier to use, so please let me know if you think any part of the ported code looks unduly messy, or can be improved.

-- AJ

**Comment by parambirs** [comment]
Looks good.

It seems that wai-routes uses some sort of convention in that the method `postSubmitR` is used to handle POST requests for the route named `SubmitR`.

**Comment by ajnsit** [comment]
Yep exactly right. Though it's not really a convention but required by the system.

So if your method is named FOO (Could be anything, including actually "FOO") and the route is called BAR (The Camel casing and ending with an R is not mandatory, just a convention), the corresponding route handler must be called fooBAR. As the example shows, you can simply leave the http method field blank and the handler will be called handleBAR instead.
